pkg/reconciler/ingress/ingress.go

```diff
@@ -324,10 +345,38 @@ func (r *Reconciler) reconcileRouteDeletion(ctx context.Context, ia *v1alpha1.In
 		return err
 	}
 
+	ingresses, err := r.ingressLister.Ingresses(ia.GetNamespace()).List(labels.SelectorFromSet(map[string]string{}))
+	if err != nil {
+		return fmt.Errorf("failed to list ingress in %q: %w", ia.GetNamespace(), err)
+	}
+
+	// If the Ingress is the last one in the namespace, remove its namespace from SMMR.
+	if len(ingresses) == 1 {
+		// In order to double check that we are reconciling proper ingress check with name and namespace
+		if ia.GetNamespace() == ingresses[0].Namespace && ia.GetName() == ingresses[0].Name {
```

Comment:
The namespace check is redundant, you can fold the name check into the first if, i.e.

```
if len(ingresses) == 1 && ia.GetName() == ingresses[0].Name {
```